### PR TITLE
Implement getWaypointForGrid(grid) & getSafeNeighboringGrids(grid)

### PR DESCRIPTION
**Pull request by dianapvalles**

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -497,14 +455,160 @@ function locationToArray(){
...
+
+  new google.maps.DirectionsRenderer({
+    map: map,
+    directions: result,
+    routeIndex: routeIndex
+  });    
+}
+
+function getWaypointForGrid(grid) {
+  fetch('/grids?requestRow=' + grid.row + '&requestCol=' + grid.col)
```
Comment (wait321) [inline portfolio/src/main/webapp/script.js:511]: fetch is an asynchronous call that doesn't block. you need to await here to block on the result. Otherwise it returns nothing immediately. I believe the function getWaypointForGrid also needs to have the `async` keyword

response = await fetch(...);`
waypoint = await response.json();
return {
      location: new google.maps.LatLng(waypoint.lat, waypoint.lng),
      stopover: true
    }

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -497,14 +455,160 @@ function locationToArray(){
...
+  let safeGridDown;
+  let safeGridRight;
+  let safeGridLeft;
+  let index = 0;
+
+  if(grid.row + 1 < 6){
+    upperGridRow = grid.row + 1;
+
+    // Might need to change the parameters once the class for the numberOfReports is merged.
+    fetch('/numberOfReports?row=' + upperGridRow + '&col=' + grid.col)
```
Comment (wait321) [inline portfolio/src/main/webapp/script.js:544]: same here. this will not block without await. you should write some dummy code to make sure it works

var grid = new Object();
grid.row = 1;
grid.col = 2;
console.log(getSafeNeightboringGrids(grid));

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (dianapvalles) [reply portfolio/src/main/webapp/script.js:511]: Oh, I used hard-coded values to test it and it worked but I will change it to avoid future problems with that, thank you!

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (dianapvalles) [reply portfolio/src/main/webapp/script.js:544]: Okay, I'll add it, thank you Yi-Tao!

File: portfolio/src/main/webapp/waypoints.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+// limitations under the License.
+
+var grid = new Object();
+grid.row = 1;
+grid.col = 2;
+
+async function getWaypointForGrid(grid) {
+  try{
+    let response = await fetch('/grids?requestRow=' + grid.row + '&requestCol=' + grid.col);
+    let waypoint =  response.json();
```
Comment (wait321) [inline portfolio/src/main/webapp/waypoints.js:22]: `await response.json()`

File: portfolio/src/main/webapp/index.html [hunk 4]
```diff
@@ -5,6 +5,7 @@
     <title>Capstone Project</title>
     <link rel="stylesheet" href="style.css">
     <script src="script.js" defer></script>
+    <script src="waypoints.js" defer></script>
```
Comment (wait321) [inline portfolio/src/main/webapp/index.html:8]: this should go above script.js since the methods in there will call the ones in waypoints.js

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -494,49 +494,7 @@ function route() {
...
-
-      console.log("Chosen route index: " + routeIndex);
-      /*directionsDisplay.setDirections(result);*/
-
-      new google.maps.DirectionsRenderer({
-        map: map,
-        directions: result,
-        routeIndex: routeIndex
-      });
+      pickSafestRoute(result);
```
Comment (wait321) [inline portfolio/src/main/webapp/script.js:497]: revert the changes in this file since salvador should be committing this part (unless I'm mistaken and he won't have this change)

File: portfolio/src/main/webapp/waypoints.js (hunk 3 above)
Comment (dianapvalles) [reply portfolio/src/main/webapp/waypoints.js:22]: Noted, thanks!

File: portfolio/src/main/webapp/index.html (hunk 4 above)
Comment (dianapvalles) [reply portfolio/src/main/webapp/index.html:8]: Noted, thanks!

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (dianapvalles) [reply portfolio/src/main/webapp/script.js:497]: Okay, I'll put the original part there then, thank you!
